### PR TITLE
remove all leftover data-hooks

### LINKED ISSUE #1586: Rip out all of the data-hooks
I don't think this is an issue yet...

Lots of legacy hooks from our Ampersand days. Let's rip them out.

nb. there's one CSS reference to a data-hook, so make sure to handle that.

### PR DESCRIPTION
**Pull request by chuckharmston**
Copying from https://github.com/mozilla/testpilot/pull/1933 to diagnose missing CI.

---

I got really sick of looking at these things. Fixes #1586

NB. There are two places where these stayed in: 
- React uses a data-hook to mount the app
- in the email component, a data-hook is used in a test

### CODE REVIEW

File: frontend/src/app/components/EmailDialog.js [hunk 1]
```diff
@@ -17,32 +17,32 @@ export default class EmailDialog extends React.Component {
...
             <div className="modal-actions">
-              <button onClick={e => this.submit(e)} data-hook="submit-email" data-l10n-id="emailOptInButton" className="submit button large default">Sign me up</button>
-              <a onClick={e => this.skip(e)} data-hook="cancel-modal" data-l10n-id="emailOptInSkip" className="cancel modal-escape" href="">Skip</a>
+              <button onClick={e => this.submit(e)} data-l10n-id="emailOptInButton" className="submit button large default">Sign me up</button>
+              <a onClick={e => this.skip(e)} data-l10n-id="emailOptInSkip" className="cancel modal-escape" href="">Skip</a>
             </div>
           </form>
         </div>}
-        {!isFirstPage && <div data-hook="second-page" className="modal">
+        {!isFirstPage && <div className="modal">
```
Comment (jaredhirsch) [inline frontend/src/app/components/EmailDialog.js]: How about an `id="second-page"` for readability? (I think the ID would help more here than in the first-page case, but for symmetry, might be better to keep or lose both.)

File: frontend/src/app/components/EmailDialog.js [hunk 2]
```diff
@@ -17,32 +17,32 @@ export default class EmailDialog extends React.Component {
...
           <header>
             <h3 className="title" data-l10n-id="emailOptInConfirmationTitle">Email Sent</h3>
           </header>
           <div className="modal-content centered">
             <div className="envelope"></div>
             <p data-l10n-id="emailOptInSuccessMessage2">Thank you!</p>
-            <button onClick={e => this.continue(e)} className="button default" data-hook="continue" data-l10n-id="emailOptInConfirmationClose">On to the experiments...</button>
+            <button onClick={e => this.continue(e)} className="button default" data-l10n-id="emailOptInConfirmationClose" data-hook="continue">On to the experiments...</button>
```
Comment (jaredhirsch) [inline frontend/src/app/components/EmailDialog.js]: It looks like the data-hook slipped by here. Please remove it.

File: frontend/src/app/components/EmailDialog.js [hunk 3]
```diff
@@ -17,32 +17,32 @@ export default class EmailDialog extends React.Component {
 
     return (
       <div className="modal-container">
-        {isFirstPage && <div data-hook="first-page" className="modal feedback-modal modal-bounce-in">
+        {isFirstPage && <div className="modal feedback-modal modal-bounce-in">
```
Comment (jaredhirsch) [inline frontend/src/app/components/EmailDialog.js]: How about an `id="first-page"` for readability?

File: frontend/src/app/components/ExperimentRowCard.js [hunk 4]
```diff
@@ -22,7 +22,7 @@ export default class ExperimentRowCard extends React.Component {
     const isCompleted = isAfterCompletedDate(experiment);
 
     return (
-      <Link data-hook="show-detail" to={`/experiments/${slug}`} onClick={(evt) => this.openDetailPage(evt)}
+      <Link to={`/experiments/${slug}`} onClick={(evt) => this.openDetailPage(evt)}
```
Comment (jaredhirsch) [inline frontend/src/app/components/ExperimentRowCard.js]: `id="show-detail"`?

File: frontend/src/app/components/Header.js [hunk 5]
```diff
@@ -31,21 +31,21 @@ export default class Header extends React.Component {
   renderSettingsMenu() {
     if (this.shouldRenderSettingsMenu()) {
       return (
-        <div data-hook="settings">
-          <div className="settings-contain" data-hook="active-user">
+        <div>
```
Comment (jaredhirsch) [inline frontend/src/app/components/Header.js]: `id="settings"`?

File: frontend/src/app/containers/ExperimentPage.js [hunk 6]
```diff
@@ -265,7 +265,7 @@ export class ExperimentDetail extends React.Component {
 
         <View {...this.props}>
 
-        {!hasAddon && <section data-hook="testpilot-promo">
+        {!hasAddon && <section>
```
Comment (jaredhirsch) [inline frontend/src/app/containers/ExperimentPage.js]: `id="testpilot-promo"`?

File: frontend/src/app/containers/ExperimentPage.js [hunk 7]
```diff
@@ -285,48 +285,48 @@ export class ExperimentDetail extends React.Component {
...
+                {subtitle && <h4 className="subtitle" data-l10n-id={this.l10nId('subtitle')}>{subtitle}</h4>}
               </header>
               { this.renderExperimentControls() }
               { this.renderMinimumVersionNotice(title, hasAddon, min_release) }
             </div>
           </div>
           <div className="sticky-header-sibling" style={{ height: `${stickyHeaderSiblingHeight}px` }} ></div>
 
-          <div data-hook="details">
+          <div>
```
Comment (jaredhirsch) [inline frontend/src/app/containers/ExperimentPage.js]: `id="details"`?

File: frontend/src/app/containers/HomePageNoAddon.js [hunk 8]
```diff
@@ -16,7 +16,7 @@ export default class HomePageNoAddon extends React.Component {
     const currentExperiments = experiments.filter(x => !isAfterCompletedDate(x));
 
     return (
-      <section data-hook="landing-page">
+      <section>
```
Comment (jaredhirsch) [inline frontend/src/app/containers/HomePageNoAddon.js]: `id="landing-page"`?

File: frontend/src/app/containers/NotFoundPage.js [hunk 9]
```diff
@@ -8,7 +8,7 @@ export default class NotFoundPage extends React.Component {
     return (
       <View centered={true} showNewsletterFooter={false} showFooter={false} showHeader={false}
             {...this.props}>
-        <div data-hook="no-found-page">
+        <div>
```
Comment (jaredhirsch) [inline frontend/src/app/containers/NotFoundPage.js]: `id="not-found-page"`?

**Review by jaredhirsch — CHANGES_REQUESTED** [review]
Hey @johngruen - Looks good overall. Couple minor things:

First, if we're going to this much trouble to remove data-hooks, I think it makes sense to remove the last two occurrences as well: a selector that React can use to bootstrap the app, and a selector used by a test. I bet that replacing the data-hook with an ID would work in both cases. I do think these last two data-hooks should be removed before landing this PR.

Second, in some cases, I think it'd be good to keep the data-hook's content, just to keep things readable/scannable. Especially when the data-hook was in the topmost node returned by a component, it's just nice to have when scanning the DOM later. I've called out spots where I think this change would help, but because readability is more of a judgment call, consider those comments to be suggestions--feel free to disregard if you don't agree.

Feel free to merge this yourself after making changes, or ping me if you want a followup review. Thanks!

**Comment by johngruen** [comment]
@6a68 yeah good points above. I'll work back through this after we land the various last minute l10n issues today

**Comment by johngruen** [comment]
@6a68 changes made. Could you give it one more look and merge since i touched a lot of files.

Cheers!

**Comment by johngruen** [comment]
Rebased!

**Comment by jaredhirsch** [comment]
lgtm, merging :+1:

**Pull request merged**